Neos.ContentRepository.Core/Classes/Service/ContentRepositoryMaintainer.php

```diff
@@ -0,0 +1,200 @@
+            $bootResult = $this->subscriptionEngine->boot();
+            if ($bootResult->errors !== null) {
+                return self::createErrorForReason('initial catchup', $bootResult->errors);
+            }
+        }
+        return null;
+    }
+
+    public function eventStoreStatus(): EventStoreStatus
+    {
+        return $this->eventStore->status();
+    }
+
+    public function subscriptionStatuses(): SubscriptionStatuses
+    {
+        return $this->subscriptionEngine->subscriptionStatuses();
+    }
+
+    public function replayProjection(SubscriptionId $subscriptionId, \Closure|null $progressCallback = null): Error|null
+    {
+        $resetResult = $this->subscriptionEngine->reset(SubscriptionEngineCriteria::create([$subscriptionId]));
+        if ($resetResult->errors !== null) {
+            return self::createErrorForReason('reset', $resetResult->errors);
+        }
+        $bootResult = $this->subscriptionEngine->boot(SubscriptionEngineCriteria::create([$subscriptionId]), $progressCallback);
+        if ($bootResult->errors !== null) {
+            return self::createErrorForReason('catchup', $bootResult->errors);
+        }
+        return null;
+    }
+
+    public function replayAllProjections(\Closure|null $progressCallback = null): Error|null
+    {
+        $resetResult = $this->subscriptionEngine->reset();
+        if ($resetResult->errors !== null) {
+            return self::createErrorForReason('reset', $resetResult->errors);
+        }
+        $bootResult = $this->subscriptionEngine->boot(progressCallback: $progressCallback);
+        if ($bootResult->errors !== null) {
+            return self::createErrorForReason('catchup', $bootResult->errors);
+        }
+        return null;
+    }
+
+    /**
+     * Catchup one specific projection.
+     *
+     * The explicit catchup is required for new projections in the booting state.
+     *
+     * We don't offer an API to catch up all projections catchupAllProjection as we would have to distinct between booting or catchup if its active already.
+     *
+     * This method is only needed in rare cases for debugging or after installing a new projection or fixing its errors.
+     */
+    public function catchupProjection(SubscriptionId $subscriptionId, \Closure|null $progressCallback = null): Error|null
+    {
+        $bootResult = $this->subscriptionEngine->boot(SubscriptionEngineCriteria::create([$subscriptionId]), progressCallback: $progressCallback);
+        if ($bootResult->errors !== null) {
+            return self::createErrorForReason('catchup', $bootResult->errors);
+        }
+        if ($bootResult->numberOfProcessedEvents > 0) {
+            // the projection was bootet
+            return null;
+        }
+        // todo the projection was active, and we might still want to catch it up ... find reason for this? And combine boot and catchup?
```

Comment:
todo discuss `catchupProjection` dilemma

Reply:
we discussed that we rather want an explicit reactivate in the subscription engine than that `cr:setup` buts the subscriptions into booting.

see https://github.com/neos/neos-development-collection/pull/5378/commits/800fd5317d2b0fdd28e008ff6c66eb2d13b97498

that will be fully implemented in the main pr.